### PR TITLE
JAMES-3105 Sending mail via JMAP: Add test for Outbox & Sent counters

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[62bab3269ceef65c1741fbbf243e5c109307ad8c] [ERROR] Failures: 
[62bab3269ceef65c1741fbbf243e5c109307ad8c] [ERROR]   CassandraSetMessagesMethodTest>SetMessagesMethodTest.sendingAMailShouldLeadToAppropriateMailboxCounters:1106 1 expectation failed.
[62bab3269ceef65c1741fbbf243e5c109307ad8c] JSON path [0][1].list[0].totalMessages doesn't match.
[62bab3269ceef65c1741fbbf243e5c109307ad8c] Expected: <1>
[62bab3269ceef65c1741fbbf243e5c109307ad8c]   Actual: 2
```

File: server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SetMessagesMethodTest.java [hunk 1]
```diff
@@ -1047,6 +1050,63 @@ public void setMessagesShouldReturnCreatedMessageWhenSendingMessage() {
...
+                "  \"properties\" : [\"unreadMessages\", \"totalMessages\"]}, " +
+                "\"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxes"))
+            .body(FIRST_MAILBOX + ".totalMessages",equalTo(0))
+            .body(FIRST_MAILBOX + ".unreadMessages", equalTo(0));
+        given()
```
Comment (Arsnael) [inline server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SetMessagesMethodTest.java]: Line break

File: server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SetMessagesMethodTest.java [hunk 2]
```diff
@@ -1047,6 +1050,66 @@ public void setMessagesShouldReturnCreatedMessageWhenSendingMessage() {
             ;
     }
 
+    @Category(BasicFeature.class)
+    @Test
+    public void sendingAMailShouldLeadToAppropriateMailboxCounters() {
```
Comment (mbaechler) [inline server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SetMessagesMethodTest.java]: it's actually two tests

**Comment by chibenwa** [comment]
Reliably failing on CI ^^

```
[503612a0aac324de37cf4a7f73b0a757fec6cc57] [ERROR]   CassandraSetMessagesMethodTest>SetMessagesMethodTest.sendingAMailShouldLeadToAppropriateMailboxCounters:1109 1 expectation failed.
[503612a0aac324de37cf4a7f73b0a757fec6cc57] JSON path [0][1].list[0].totalMessages doesn't match.
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Expected: <1>
[503612a0aac324de37cf4a7f73b0a757fec6cc57]   Actual: 2
```

**Comment by chibenwa** [comment]
```
Request method:	POST
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Request URI:	http://localhost:39421/jmap
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Proxy:			<none>
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Request params:	<none>
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Query params:	<none>
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Form params:	<none>
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Path params:	<none>
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Headers:		Accept=application/json, application/javascript, text/javascript, text/json
[503612a0aac324de37cf4a7f73b0a757fec6cc57] 				Authorization=5a6d2274-a6be-4d0d-a225-488d3683baee
[503612a0aac324de37cf4a7f73b0a757fec6cc57] 				Content-Type=application/json; charset=UTF-8
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Cookies:		<none>
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Multiparts:		<none>
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Body:
[503612a0aac324de37cf4a7f73b0a757fec6cc57] [
[503612a0aac324de37cf4a7f73b0a757fec6cc57]     [
[503612a0aac324de37cf4a7f73b0a757fec6cc57]         "getMailboxes",
[503612a0aac324de37cf4a7f73b0a757fec6cc57]         {
[503612a0aac324de37cf4a7f73b0a757fec6cc57]             "ids": [
[503612a0aac324de37cf4a7f73b0a757fec6cc57]                 "b4504ed0-7255-11ea-a767-5944437abfac"
[503612a0aac324de37cf4a7f73b0a757fec6cc57]             ],
[503612a0aac324de37cf4a7f73b0a757fec6cc57]             "properties": [
[503612a0aac324de37cf4a7f73b0a757fec6cc57]                 "unreadMessages",
[503612a0aac324de37cf4a7f73b0a757fec6cc57]                 "totalMessages"
[503612a0aac324de37cf4a7f73b0a757fec6cc57]             ]
[503612a0aac324de37cf4a7f73b0a757fec6cc57]         },
[503612a0aac324de37cf4a7f73b0a757fec6cc57]         "#0"
[503612a0aac324de37cf4a7f73b0a757fec6cc57]     ]
[503612a0aac324de37cf4a7f73b0a757fec6cc57] ]
[503612a0aac324de37cf4a7f73b0a757fec6cc57] 
[503612a0aac324de37cf4a7f73b0a757fec6cc57] HTTP/1.1 200 OK
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Access-Control-Allow-Origin: *
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Access-Control-Allow-Methods: GET, POST, DELETE, PUT
[503612a0aac324de37cf4a7f73b0a757fec6cc57] Access-Control-Allow-Headers: Content-Type, Authorization, Accept
[503612a0aac324de37cf4a7f73b0a757fec6cc57] content-type: application/json
[503612a0aac324de37cf4a7f73b0a757fec6cc57] content-length: 160
[503612a0aac324de37cf4a7f73b0a757fec6cc57] 
[503612a0aac324de37cf4a7f73b0a757fec6cc57] [
[503612a0aac324de37cf4a7f73b0a757fec6cc57]     [
[503612a0aac324de37cf4a7f73b0a757fec6cc57]         "mailboxes",
[503612a0aac324de37cf4a7f73b0a757fec6cc57]         {
[503612a0aac324de37cf4a7f73b0a757fec6cc57]             "accountId": null,
[503612a0aac324de37cf4a7f73b0a757fec6cc57]             "state": null,
[503612a0aac324de37cf4a7f73b0a757fec6cc57]             "list": [
[503612a0aac324de37cf4a7f73b0a757fec6cc57]                 {
[503612a0aac324de37cf4a7f73b0a757fec6cc57]                     "id": "b4504ed0-7255-11ea-a767-5944437abfac",
[503612a0aac324de37cf4a7f73b0a757fec6cc57]                     "totalMessages": 2,
[503612a0aac324de37cf4a7f73b0a757fec6cc57]                     "unreadMessages": 2
[503612a0aac324de37cf4a7f73b0a757fec6cc57]                 }
[503612a0aac324de37cf4a7f73b0a757fec6cc57]             ],
[503612a0aac324de37cf4a7f73b0a757fec6cc57]             "notFound": null
[503612a0aac324de37cf4a7f73b0a757fec6cc57]         },
[503612a0aac324de37cf4a7f73b0a757fec6cc57]         "#0"
[503612a0aac324de37cf4a7f73b0a757fec6cc57]     ]
[503612a0aac324de37cf4a7f73b0a757fec6cc57] ]
```

**Comment by chibenwa** [comment]
(what if the move was "incrementing" instead of decrementing?)

**Comment by chibenwa** [comment]
We need a await on the final conditions BTW

**Comment by chibenwa** [comment]
```
[014feb13073531473896b427ff6ee1ea90631555] [INFO] There is 1 error reported by Checkstyle 8.29 with checkstyle.xml ruleset.
[014feb13073531473896b427ff6ee1ea90631555] [ERROR] src/main/java/org/apache/james/jmap/draft/send/PostDequeueDecoratorFactory.java:[34,8] (imports) UnusedImports: Unused import - org.apache.james.queue.api.RawMailQueueItemDecoratorFactory.
```

**Comment by chibenwa** [comment]
test this please

**Comment by rouazana** [comment]
> Do we have an alternative design to replace this decorator entirely?

It won't be necessary anymore with new JMAP spec: https://www.rfc-editor.org/rfc/rfc8621.html#section-7.5.1

**Comment by chibenwa** [comment]
> Do we have an alternative design to replace this decorator entirely?

Yes. 

A `onSuccess` pipeline for RemoteDelivery, and configuration after LocalDelivery, along with `MessageSubmission JMAP datamodel`

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**